misc/bazel/pkg.bzl

```diff
@@ -238,25 +264,37 @@ def codeql_pack(
             kind = kind,
             visibility = ["//visibility:private"],
         )
-        pkg_zip(
-            name = internal(kind + "-zip-base"),
-            srcs = [internal(kind)],
-            visibility = ["//visibility:private"],
-        )
-        _zipmerge(
-            name = internal(kind + "-zip"),
-            base = internal(kind + "-zip-base"),
+        if zips:
+            pkg_zip(
+                name = internal(kind + "-zip-base"),
+                srcs = [internal(kind)],
+                visibility = ["//visibility:private"],
+                compression_level = compression_level,
+            )
+            _zipmerge(
+                name = internal(kind + "-zip"),
+                base = internal(kind + "-zip-base"),
+                zips = zips,
+                zip_name = zip_filename,
+                kind = kind,
+                prefix = name,
+                visibility = visibility,
+            )
+        else:
+            pkg_zip(
+                name = internal(kind + "-zip"),
+                srcs = [internal(kind)],
+                visibility = ["//visibility:private"],
+                package_dir = name,
+                package_file_name = name + "-" + (_detect_platform() if kind == "arch" else "generic") + ".zip",
```

Comment:
Can we pull out this logic into a variable in `codeql_pack`, and remove the same/similar filename scheme from `_zipmerge`? That'd make `zipmerge` a little bit more generic, too.